### ISSUE TITLE
Lifting the 4D limitation from the input_data masker tools

### ISSUE THREAD

**Issue by dangom** [nilearn/nilearn#1923]
From the [documentation](https://nilearn.github.io/manipulating_images/masker_objects.html#nifti-masker):
> In any analysis, the first step is to load the data. It is often convenient to apply some basic data transformations and to turn the data in a 2D (samples x features) matrix, where the samples could be different time points, and the features derived from different voxels (e.g., restrict analysis to the ventral visual stream), regions of interest (e.g., extract local signals from spheres/cubes), or pre-specified networks (e.g., look at data from all voxels of a set of network nodes). Think of masker objects as swiss-army knifes for shaping the raw neuroimaging data in 3D space into the units of observation relevant for the research questions at hand.

**What would you like changed/added and why?**

The Masker family of tools that nilearn provides is really awesome, but it is limited to 4D input files. I don't see why it shouldn't be able to handle 3D files as well, in which case the data in a 2D (samples x features) matrix would have dimension (1 x features).

**What would be the benefit? Does the change make something easier to use?**

Yes. Say I have a statistical image and I would like to see the mean activation within ROIs of a mask/parcellation. Currently that's not possible with masker because inputs have to be 4D. If we lift this limitation, than that'd be as easy as:
```python
from nilearn import datasets, input_data, image

mt = datasets.fetch_neurovault_motor_task()
parcels = datasets.fetch_atlas_harvard_oxford('cort-maxprob-thr25-2mm')

data = image.load_img(mt.images[0])
target = image.load_img(parcels["maps"])

# Mean of data in each parcel of the atlas? Potentially useful when we have a labelled atlas.
masker = input_data.NiftiLabelsMasker(target)
results = masker.fit_transform(data)  # Fails because data is not 4D.
```

The current hackish workaround is:
```python
results = masker.fit_transform(image.concat_img([data, data]))[0,:]
```

**Comment by GaelVaroquaux** [comment]
Hi Daniel,

Thanks for the suggestion, but I would rather keep the restriction. What
you are proposing amounts really to a different semantic. Having objects
too versatile will make them hard to understand for end-users and hard to
maintain.

To give a specific example of what I mean, our code that deals with
checking that input files have the same 3D vs 4D dimensionality, and of
casting inputs like list of 3D files in 4D representations, is actually
quite hairy.

**Comment by dangom** [comment]
Indeed some parameters of the masker don't make sense for 3D images, such as the TR, low and high pass filters, and standardize. In that I agree with you that the semantics are different.

On the other hand, the maskers are super convenient in that they take care of resampling and checking for shapes and affines. Currently no similar functionality exists for masking 3D files in nilearn, afaik.

**Comment by bthirion** [comment]
Why not simply use 
```
data = image.load_img(mt.images[:1])
```
or [mt.images[0]]
I agree that this is slightly less intuitive, yet not a big deal.

**Comment by dangom** [comment]
Wow, I didn't expect that to work, but it does and is definitely cleaner than my workaround.
Thanks a lot for the quick response!
